src/test/scala/mesosphere/marathon/upgrade/DeploymentActorTest.scala

```diff
@@ -280,4 +279,78 @@ class DeploymentActorTest
       system.shutdown()
     }
   }
+
+  test("Scale with tasksToKill") {
+    implicit val system = ActorSystem("TestSystem")
+    val managerProbe = TestProbe()
+    val receiverProbe = TestProbe()
+    val app1 = AppDefinition(id = PathId("app1"), cmd = Some("cmd"), instances = 2, version = Timestamp(0))
+    val origGroup = Group(PathId("/foo/bar"), Set(app1))
+
+    val app1New = app1.copy(instances = 1, version = Timestamp(1000))
+
+    val targetGroup = Group(PathId("/foo/bar"), Set(app1New))
+
+    val task1_1 = MarathonTasks.makeTask("task1_1", "", Nil, Nil, app1.version).toBuilder.setStartedAt(0).build()
+    val task1_2 = MarathonTasks.makeTask("task1_2", "", Nil, Nil, app1.version).toBuilder.setStartedAt(500).build()
+    val task1_3 = MarathonTasks.makeTask("task1_3", "", Nil, Nil, app1.version).toBuilder.setStartedAt(1000).build()
+
+    val plan = DeploymentPlan(original = origGroup, target = targetGroup, toKill = Map(app1.id -> Set(task1_2)))
+
+    when(tracker.get(app1.id)).thenReturn(Set(task1_1, task1_2, task1_3))
+
+    // the AppDefinition is never used, so it does not mater which one we return
+    when(repo.store(any())).thenReturn(Future.successful(AppDefinition()))
+
+    when(driver.killTask(TaskID(task1_1.getId))).thenAnswer(new Answer[Status] {
```

Comment:
Indeed. There are even more flaws:
- the original AppDefinition should have an instance count of 3, since 3 tasks are 'running'
- the new AppDefinition should have an instance count of 2, since we kill 1 task and scale